x-pack/plugins/apm/server/routes/services/annotations/index.ts

```diff
@@ -41,6 +41,9 @@ export async function getServiceAnnotations({
     searchAggregatedTransactions,
     start,
     end,
+  }).catch(() => {
```

Comment:
FWIW, this is not the same behaviour as before. If we don't find stored annotations, and want to use derived annotations, this should result in an error, but with an empty catch handler, it no longer does.

Reply:
ah, I see your point.

Reply:
> this should result in an error

What do you mean with data? isn't that what we're trying to prevent by fixing this issue?

Reply:
The issue arises when the request is aborted and the promise is not caught, because the catch handler has not been added. This happens when the request for the stored annotations has not finished, but the request for derived annotations has started. What we want is for that situation to be resolved, but if an error for another reason occurs (e.g. the request to ES times out), and there are no stored annotations, the error should be propagated to the route handler.

Reply:
What about now? https://github.com/elastic/kibana/pull/135752/commits/5e704eb75a8486eea38003f76df468a2391983a7